daemon.go

```diff
@@ -201,3 +207,23 @@ func (d *Daemon) Close() error {
 
 	return merr.ErrorOrNil()
 }
+
+func (d *Daemon) awaitTermination() {
+	go func() {
+		d.terminateWG.Wait()
+		time.Sleep(time.Second * 30)
+		d.Close()
+	}()
+}
+
+func (d *Daemon) KillOnTimeout(timeout time.Duration) {
```

Comment:
```suggestion
func (d *Daemon) KillOnTimeout(timeout time.Duration) {
```

In constrast to `awaitTermination()`, I think it is better to move `go ...` inside this function since it is a public interface of the daemon. A user is likely to call `KillOnTimeout(timeout)` and expect that this just saves a setting for the daemon (not blocks).